### PR TITLE
[DataLake]Add dev_requirements.txt

### PR DESCRIPTION
**Pull request by xiafu-msft**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-file-datalake/dev_requirements.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+-e ../../../tools/azure-devtools
+-e ../../../tools/azure-sdk-tools
+../../core/azure-core
+../azure-storage-blob
+-e ../../identity/azure-identity
```
Comment (rakshith91) [inline sdk/storage/azure-storage-file-datalake/dev_requirements.txt]: is this really required?

Comment (xiafu-msft) [reply sdk/storage/azure-storage-file-datalake/dev_requirements.txt]: I think we want this to test oauth authentication(eg. we want to import ClientSecretCredential from azure-identity) not sure why blob and queue have identity while fhileshare doesn't have this 
https://github.com/Azure/azure-sdk-for-python/blob/8380bfd325f500aaa2999ac2f96be9f85329bc39/sdk/storage/azure-storage-file-share/dev_requirements.txt#L1-L5

File: sdk/storage/azure-storage-file-datalake/dev_requirements.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+-e ../../../tools/azure-devtools
```
[the remark this answers is not in the document]
Comment (rakshith91) [reply sdk/storage/azure-storage-file-datalake/dev_requirements.txt]: -e installs the "editable" version..it means it installs whatever the changes are in your repository,

File: sdk/storage/azure-storage-file-datalake/dev_requirements.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+-e ../../../tools/azure-sdk-tools
+../../core/azure-core
+../azure-storage-blob
```
[the remark this answers is not in the document]
Comment (rakshith91) [reply sdk/storage/azure-storage-file-datalake/dev_requirements.txt]: >=12.0.0 is a good idea IMO.

File: sdk/storage/azure-storage-file-datalake/dev_requirements.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (xiafu-msft) [reply sdk/storage/azure-storage-file-datalake/dev_requirements.txt]: is `../../../tools/azure-devtools` also doing the same thing?

File: sdk/storage/azure-storage-file-datalake/dev_requirements.txt (hunk 1 above)
Comment (lmazuel) [reply sdk/storage/azure-storage-file-datalake/dev_requirements.txt]: This file is a dev file, if you have tests that uses identity, then it needs to be here. This file doesn't impact runtime dependency of the setup.py

File: sdk/storage/azure-storage-file-datalake/dev_requirements.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (lmazuel) [reply sdk/storage/azure-storage-file-datalake/dev_requirements.txt]: keep it a relative link for this file (not runtime dep)

**Pull request merged**